### DISCUSSION TITLE
Allow or deny interface list

### DISCUSSION THREAD

**Discussion by advancenXX** [eProsima/Fast-DDS#3941] [Q&A] [answered]
I find that RTI has allow/deny_interfaces_list, and I am wondering if we have mechanism to set interface that allowed.
I have Participant A with 3 interfaces, so Participant B may send PDP/EDP to those three interfaces.
Is there any way to limit B to send packet to the only one interface I want.

Thanks a lot.

**Comment by JesusPoderoso** [comment]
Hi @advancenXX, you can configure the Participant transport with the IP addresses that belongs to the desired interfaces [as follows](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/whitelist.html).

**Reply by advancenXX** [reply]
Hi @JesusPoderoso, thanks for your reply.
The way you mentioned is to be set at Participant A's side, but what I want is to limit it at Participant B.
        A ------interface 1-----> B
        A ------interface 2-----> C
For particiapnt B, we want to avoid A's interface 2 when sending PDP or EDP.

**Reply by JesusPoderoso** [reply]
It applies in the same way, as long as you configure both locator and interface whitelist with the same interface IP address. At the end of the day, the locator is configuring the IP where the participant is listening.
Lets say that A, B and C have different IP addresses in interface 1 and 2:
| - | A | B | C |
|---|---|---|---|
| I1 | 1.1.1.10 | 1.1.1.11 | 1.1.1.12 |
| I2 | 2.2.2.20 | 2.2.2.21 | 2.2.2.22 |

You can configure A with two locators, one for each interface (I1: IP 1.1.1.10 and port 7441 for instance, and I2: IP 2.2.2.20 and port 7451), and include both IPs in the interface whitelist. That would allow A to receive data in both interfaces.
In B case, you only configure one locator in the I1 interface (IP 1.1.1.11 and port 7442 for instance), and include that I1 IP in the interface whitelist.
In that way, B will not be listening in its assigned IP address 2.2.2.21 in I2.

**Reply by advancenXX** [reply]
Well, in that way when B receives A's PDP message, B may send two EDP packets to A like,
        1.1.1.11  ==> 1.1.1.10
        1.1.1.11  ==> 2.2.2.20
The second EDP packet may cause some route problem.
Is there any way to limit the second one?

**Reply by JesusPoderoso** [reply]
Hi @advancenXX , now I understand your concern and you are right.  Currently there is no way to perform the limitation you are asking for, but that feature is WIP and will be released in next Fast DDS release v2.13.0. Hope that helps!

**Reply by juanlofer-eprosima** [reply, marked as answer]
Hi @advancenXX , the new feature mentioned by @JesusPoderoso was introduced in v2.14.0 release. The corresponding documentation is available [here](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/interfaces.html#netmask-filtering). Feel free to give it a shot.
